naxsi_src/naxsi_utils.c

```diff
@@ -653,25 +653,21 @@ ngx_http_wlr_finalize_hashtables(ngx_conf_t *cf, ngx_http_dummy_loc_conf_t  *dlc
     }
   }
 
-  FILE *file;
-  char fname[1024];
+  FILE *file=NULL;
+  char *fname=NULL;
   if(dlc->whitelist_file)
   {
-    size_t fname_size = dlc->whitelist_file->len;
-    if (fname_size >= sizeof(fname)) {
-      fname_size = sizeof(fname);
-    }
-    memcpy(fname, dlc->whitelist_file->data, fname_size);
+    fname=malloc(strlen((char *)dlc->whitelist_file->data)*sizeof(char)+1);
```

Comment:
i think a `fname = calloc(1024, sizeof(char))` is enough and you need to check also that `fname` is a valid pointer (not NULL).

Reply:
Corrected